src/cluster_legacy.c

```diff
@@ -1049,13 +1049,22 @@ int clusterSaveConfig(int do_fsync) {
     return retval;
 }
 
+/* Save the cluster configuration file. If the save fails, exit the process. */
 void clusterSaveConfigOrDie(int do_fsync) {
     if (clusterSaveConfig(do_fsync) == C_ERR) {
         serverLog(LL_WARNING, "Fatal: can't update cluster config file.");
         exit(1);
     }
 }
 
+/* Save the cluster configuration file. If the save fails, print the log. */
+void clusterSaveConfigOrLog(int do_fsync) {
+    if (clusterSaveConfig(do_fsync) == C_ERR) {
+        serverLog(LL_WARNING, "Cluster config file is applying a change even though "
+                              "it is unable to write to disk.");
```

Comment:
Slightly strange English. Maybe better:

```suggestion
        serverLog(LL_WARNING, "Cluster config updated even though writing "
                              "the cluster config file to disk failed.");
```

Reply:
thanks, it is better.